src/OpenDebugAD7/MILaunchOptions.cs

```diff
@@ -496,7 +509,17 @@ private static string GetLLDBMIPath()
                     string miDebuggerArgs = XmlSingleQuotedAttributeEncode(jsonLaunchOptions.MIDebuggerArgs);
                     xmlLaunchOptions.Append(String.Concat("  MIDebuggerArgs='", miDebuggerArgs, "'\n"));
                 }
-                xmlLaunchOptions.Append(String.Concat("  WaitDynamicLibLoad='false'\n"));
+
+                // If we get SymbolLoadInfo and an ExceptionList or LoadAll is set. We will need to have WaitDynamicLibLoad.
+                if (jsonLaunchOptions.SymbolLoadInfo != null && 
+                    (!String.IsNullOrWhiteSpace(jsonLaunchOptions.SymbolLoadInfo.ExceptionList) || jsonLaunchOptions.SymbolLoadInfo.LoadAll.HasValue))
+                {
+                    xmlLaunchOptions.Append(String.Concat("  WaitDynamicLibLoad='true'\n"));
```

Comment:
Did you test this to see if it behaves correctly?

Reply:
Tested with a program and a static library, with LoadAll set to false, it does not step into the library code. With LoadAll false and ExpectionList with the shared library name it does step in. With LoadAll true and ExceptionList set to the shared library name, it does not step in. 
    LoadAll true and exception list empty, it does what it does normally